code/game/objects/items/weapons/teleportation.dm

```diff
@@ -146,10 +153,28 @@ Frequency:
 		return
 	var/T = L[t1]
 	user.show_message("<span class='notice'>Locked In.</span>", 2)
-	var/obj/effect/portal/P = new /obj/effect/portal(get_turf(src), T, src, creation_mob = user)
+	var/obj/effect/portal/P = new /obj/effect/portal/hand_tele(get_turf(src), T, src, creation_mob = user)
 	try_move_adjacent(P)
 	active_portals++
 	add_fingerprint(user)
 
+/obj/item/hand_tele/emp_act(severity)
+	var/time = rand(25 SECONDS, 30 SECONDS) * severity
+	emp_timer = world.time + time
+	emp_update_icon()
+	addtimer(CALLBACK(src, PROC_REF(emp_update_icon)), time)
+	return ..()
+
+/obj/item/hand_tele/proc/emp_update_icon()
+	if(emp_timer - world.time > 0.1 SECONDS)
```

Comment:
Same thing here as before. Just use a boolean like `emped` or something on the hand tele itself, and let the callback handle the time stuff instead of trying to hope you get the times lined up